curvefs/src/client/s3/client_s3_cache_manager.cpp

```diff
@@ -482,10 +491,24 @@ bool FileCacheManager::ReadKVRequestFromLocalCache(const std::string &name,
                                                    uint64_t offset,
                                                    uint64_t len) {
     uint64_t start = butil::cpuwide_time_us();
+    if (!s3ClientAdaptor_->HasDiskCache()) {
+        return false;
+    }
+    if (!s3ClientAdaptor_->GetDiskCacheManager()->IsCached(name) &&
+        len >= FLAGS_bigIoSize && s3ClientAdaptor_->GetPrefetchBlocks()) {
+        int retry = 0;
+        do {
+            VLOG(6) << "wait for download object: " << name;
+            bthread_usleep(FLAGS_bigIoRetryIntervalUs);
+            if (++retry >= FLAGS_bigIoRetryTimes) {
+                LOG(WARNING) << "download object: " << name << " timeout";
+                return false;
+            }
+        } while (!s3ClientAdaptor_->GetDiskCacheManager()->IsCached(name));
```

Comment:
There are multi same operations, maybe you can use one function to replace this chain operation.

Reply:
fix